apollo-router/src/introspection.rs

```diff
@@ -27,15 +26,13 @@ static KNOWN_INTROSPECTION_QUERIES: Lazy<Vec<String>> = Lazy::new(|| {
 /// A cache containing our well known introspection queries.
 #[derive(Debug)]
 pub struct Introspection {
-    cache: RwLock<HashMap<String, Response>>,
+    cache: HashMap<String, Response>,
 }
 
 impl Introspection {
     #[cfg(test)]
     pub fn from_cache(cache: HashMap<String, Response>) -> Self {
-        Self {
-            cache: RwLock::new(cache),
-        }
+        Self { cache }
     }
 
     /// Create a `NaiveIntrospection` from a `Schema`.
```

Comment:
a part of naive introspection still lives on because we build a cache of known introspection queries. I am not sure it is still needed, but it does not hurt